auth/store.go

```diff
@@ -1159,11 +1159,12 @@ func (as *authStore) AuthInfoFromTLS(ctx context.Context) (ai *AuthInfo) {
 
 	tlsInfo := peer.AuthInfo.(credentials.TLSInfo)
 	for _, chains := range tlsInfo.State.VerifiedChains {
-		if len(chains) < 1 {
+		cn := chains[0].Subject.CommonName
+		if len(chains) < 1 || cn == "" {
 			continue
 		}
 		ai = &AuthInfo{
-			Username: chains[0].Subject.CommonName,
```

Comment:
@mitake I remember we had conversation about this `chains[0]`, but can't remember what the reasoning was :0